presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -204,8 +204,9 @@ public ThriftHiveMetastoreStats getStats()
                     .stopOnIllegalExceptions()
                     .run("getTable", stats.getGetTable().wrap(() -> {
                         try (HiveMetastoreClient client = clientProvider.createMetastoreClient()) {
-                            org.apache.hadoop.hive.metastore.api.Table table = client.getTable(databaseName, tableName);
-                            if (table.getTableType().equals(TableType.VIRTUAL_VIEW.name()) && (!isPrestoView(table))) {
+                            Table table = client.getTable(databaseName, tableName);
+                            if (table.getTableType().equals(TableType.VIRTUAL_VIEW.name()) &&
+                                    (!"true".equals(table.getParameters().get(PRESTO_VIEW_FLAG)))) {
```

Comment:
Please leave `isPrestoView` as a method. It contains the kind of logic that I would prefer to be in a special-purpose function.

BTW, I find it easier to read if there isn't a `(...)` around `!`. But I'm ok if you feel strongly otherwise.